ssl/t1_enc.c

```diff
@@ -437,8 +437,12 @@ int tls1_change_cipher_state(SSL_CONNECTION *s, int which)
         goto skip_ktls;
 
     /* ktls works with user provided buffers directly */
-    if (BIO_set_ktls(bio, &crypto_info, which & SSL3_CC_WRITE))
+    if (BIO_set_ktls(bio, &crypto_info, which & SSL3_CC_WRITE)) {
+        if ((s->options & SSL_OP_ENABLE_KTLS_TX_ZEROCOPY_SENDFILE) &&
+            (which & SSL3_CC_WRITE))
+                BIO_set_ktls_tx_zerocopy_sendfile(bio);
         SSL_set_options(SSL_CONNECTION_GET_SSL(s), SSL_OP_NO_RENEGOTIATION);
+    }
```

Comment:
Side note: This is needed right now, but all this code will be deleted in the not-too-distant future as part of my write record layer refactor work. the changes in ktls_meth.c will be sufficient.

Reply:
OK, I'm keeping it for now, feel free to remove it when it becomes unneeded.